internal/filter/filterottl/bridge.go

```diff
@@ -0,0 +1,194 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filterottl // import "github.com/open-telemetry/opentelemetry-collector-contrib/internal/filter/filterottl"
+
+import (
+	"fmt"
+	"strings"
+
+	"go.opentelemetry.io/collector/component"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/filter/expr"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/filter/filterconfig"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/filter/filterset"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottlspan"
+)
+
+const (
+	serviceNameStaticStatement        = `resource.attributes["service.name"] == "%v"`
+	spanNameStaticStatement           = `name == "%v"`
+	spanKindStaticStatement           = `kind == "%v"`
+	scopeNameStaticStatement          = `instrumentation_scope.name == "%v"`
+	scopeVersionStaticStatement       = `instrumentation_scope.version == "%v"`
+	attributesStaticStatement         = `attributes["%v"] == %v`
+	resourceAttributesStaticStatement = `resource.attributes["%v"] == %v`
+
+	serviceNameRegexStatement        = `IsMatch(resource.attributes["service.name"], "%v")`
+	spanNameRegexStatement           = `IsMatch(name, "%v")`
+	spanKindRegexStatement           = `IsMatch(kind, "%v")`
+	scopeNameRegexStatement          = `IsMatch(instrumentation_scope.name, "%v")`
+	scopeVersionRegexStatement       = `IsMatch(instrumentation_scope.version, "%v")`
+	attributesRegexStatement         = `IsMatch(attributes["%v"], "%v")`
+	resourceAttributesRegexStatement = `IsMatch(resource.attributes["%v"], "%v")`
+)
+
+func NewSpanSkipExprBridge(mc *filterconfig.MatchConfig) (expr.BoolExpr[ottlspan.TransformContext], error) {
```

Comment:
Some tests will be great

Reply:
Added some extensive tests to `filterspan_test.go` to show that the output of the bridge matches the output of filterspan's current solution.